### PR TITLE
BUG: eq not implemented for categorical and arrow backed strings

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -628,7 +629,9 @@ def __setstate__(self, state) -> None:
 
     def _cmp_method(self, other, op):
         pc_func = ARROW_CMP_FUNCS[op.__name__]
-        if isinstance(other, (ArrowExtensionArray, np.ndarray, list, BaseMaskedArray)):
+        if isinstance(
+            other, (ArrowExtensionArray, np.ndarray, list, BaseMaskedArray)
+        ) or isinstance(getattr(other, "dtype", None), CategoricalDtype):
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:634]: So for Categorical, we don't need to extract the underlying values?

Comment (phofl) [reply pandas/core/arrays/arrow/array.py:634]: That's what _box_pa is doing, isn't it?

Comment (mroeschke) [reply pandas/core/arrays/arrow/array.py:634]: True, I guess in _box_pa `pa.array` is correctly handling it luckily which is good.

**Review by mroeschke — APPROVED** [review]
Merge conflict otherwise LGTM

**Pull request merged**